addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -51,6 +51,12 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
         willUnmount() {
             this.env.pos.off('change:selectedClient', null, this);
         }
+        willStart() {
```

Comment:
I guess it's better to put this at `after_load_server_data` since you would want to call it once. Putting it in `willStart` of the ProductScreen component means that everytime ProductScreen is mounted (instantiated), this lifecycle method is called. Which is why you have to create the `productsLoading` field in the config. You won't be needing that state field when you are sure that the loading of products in the background is only called once.